pkg/apis/provisioning/v1alpha5/requirements.go

```diff
@@ -223,26 +161,30 @@ func (r Requirements) Validate() (errs error) {
 			r.hasRequirement(withKeyAndOperator(requirement.Key, v1.NodeSelectorOpExists))) {
 			errs = multierr.Append(errs, fmt.Errorf("operator %s cannot coexist with other operators for key %s", v1.NodeSelectorOpDoesNotExist, requirement.Key))
 		}
+		// Excludes cases when In and NotIn have overlaps.
```

Comment:
How were we missing this case before? Is this caught somewhere else? Isn't it acceptable to have overlap of In and NotIn, we just take the intersection?